src/it/java/org/checkstyle/plugins/sonar/CheckJarTest.java

```diff
@@ -19,22 +19,27 @@
 
 package org.checkstyle.plugins.sonar;
 
-import static org.junit.Assert.assertTrue;
+import org.junit.Test;
 
-import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
 
-import org.junit.Test;
+import static org.junit.Assert.assertFalse;
 
 public class CheckJarTest {
     private static final String VERSION = "4.30";
 
     @Test
-    public void testJarPresence() {
-        final boolean snapshotExists = new File("target/checkstyle-sonar-plugin-"
-                                                + VERSION + "-SNAPSHOT.jar").exists();
-        final boolean releaseExists = new File("target/checkstyle-sonar-plugin-"
-                                               + VERSION + ".jar").exists();
-        assertTrue("Jar should exists",
-                   snapshotExists || releaseExists);
+    public void testJarPresence() throws IOException {
+        final List<Path> files = Files.find(Paths.get("target"), 1, (path, basicFileAttributes) -> {
+            return path.toString()
+                    .matches(".*checkstyle-sonar-plugin-\\d+\\.\\d+(-SNAPSHOT)?\\.jar");
+        })
+                .collect(Collectors.toList());
```

Comment:
please move lambda out to have a variable for it.
it is very hard to read code like this.